### DISCUSSION TITLE
Better way to deploy a yarn workspaces multi-site nextjs monorepo?

### DISCUSSION THREAD

**Discussion by eivindml** [vercel/vercel#4582] [Help]
## Problem

I have project I'm working on where a website, with some tweaks should be deployed to many different domains.

I have set this up with a yarn workspace with a `core` workspace that have all shared code (components, store, hooks etc), and then a workspace for each nextjs website, which then uses the `core library`. This works great, but is difficult to deploy easily to different Vercel projects.

```
packages/
  core/ # shared components, hooks, store, utils etc.
  website1/ # next.js website which uses core library
  website2/ # next.js website which uses core library
```

## Current solution 

Current solution is that I have created `.vercel.{websiteName}` and `now.{websiteName}.json` for each of the website, and then written a script that lets me write `./deploy {websiteName}` and will then copy the corresponding folder and json file to `.vercel` and `now.json` and then run `now`. 

You can see the structure here:
<img width="195" alt="Skjermbilde 2020-06-05 kl  18 33 07" src="https://user-images.githubusercontent.com/202450/83901894-c2c87100-a75b-11ea-894e-087252578f6a.png">

It works for now, but the number of websites will grow quickly, and it's a bit cumbersome to create the correct files and folders etc. for each.

## Question

a) Is there currently any better solutions for doing this? b) And will there be any better solutions in the future?

**Comment by eivindml** [comment]
The script simply does this:

```javascript 
#! /usr/local/bin/node
const fs = require('fs-extra')
const childProcess = require('child_process')

console.log('Deploy')

const name = process.argv[2]
const flag = process.argv[3]

if (name === undefined) {
  console.log('usage:')
  return
}

const files = fs.readdirSync('./')

if (!files.includes('.vercel.' + name)) {
  console.log('.vercel.' + name + ' does not exist!')
  return
}

if (!files.includes('now.' + name + '.json')) {
  console.log('now.' + name + '.json' + ' does not exist!')
  return
}

if (files.includes('now.json')) {
  console.log('Deleting now.json')
  fs.unlinkSync('now.json')
}

if (files.includes('.vercel')) {
  console.log('Deleting .vercel')
  fs.removeSync('.vercel')
}

console.log('Copying ' + '.vercel.' + name + ' to .vercel')
fs.copySync('.vercel.' + name, '.vercel')

console.log('Copying ' + 'now.' + name + '.json to now.json')
fs.copySync('now.' + name + '.json', 'now.json')

if (flag === '--prod') {
  console.log('Deploying to production')
  childProcess.execSync('now --prod')
  return
}

console.log('Deploying to staging')
childProcess.execSync('now')
```

**Comment by itstheandre** [comment]
EDIT: Got it working for me as well. Not the cleanest setup, not the same flow as you, but at least this for now helps 👍 

Hey! Is this still working?
For now this seems like an acceptable tradeoff no?

**Reply by eivindml** [reply]
Yeah, it's working quite OK. I have now been able to share most of the code from a `core/` package. My only issue with this setup now, is that each project needs it's own next.js + react install, and I need to remember to update that to the same version for each project.
